### DISCUSSION TITLE
Is it possible to have more than one feed/service behind a single hostname?

### DISCUSSION THREAD

**Discussion by venteto** [MarshalX/atproto#224] [Questions] [answered]
Looking at the default example I suspect I can only have one feed generator service per hostname, and I would possibly have to purchase a domain name, then create multiple subdomains, and then have nginx proxy each subdomain to a separate Flask app.

But if possible I'd rather just use one free Fly.io instance and serve multiple feed generator services from just their default subdomain for my Fly account, for example, `https://my-fly-app.fly.dev`, and not have to purchase a domain name.

**Comment by MarshalX** [comment, marked as answer]
You can host as many as you want feeds using only one flask server and only one domain name. The difference will be in the parameter which feed is requested 

You have this GET route: https://github.com/MarshalX/bluesky-feed-generator/blob/2977b0f9806af99f73cf14f6d491ce91205f1765/server/app.py#L67-L81. 

Inside this route, you have the `feed` string param. That you can use to match the feed algorithm. 

To register new algorithms you should add it here: https://github.com/MarshalX/bluesky-feed-generator/blob/2977b0f9806af99f73cf14f6d491ce91205f1765/server/algos/__init__.py#L3-L5

also, you should implement another data filter for new events in the network. and store it separately in db for another algorithm. for example you can create another table in db or mark post row with name of feed. https://github.com/MarshalX/bluesky-feed-generator/blob/2977b0f9806af99f73cf14f6d491ce91205f1765/server/data_filter.py#L23

In this example algorithm equals to custom feed

**Reply by venteto** [reply]
When working with the data filter, is it possible to retrieve a user DID from the post, so the DID could be matched against a list of DID's that a feed could be limited to (like a curated user list)?

**Reply by MarshalX** [reply]
Did of the author? Who wrote new post? Or did who requested custom feed posts?

**Reply by venteto** [reply]
Sorry, the account who created the new post in the firehose.

For example, let's say I want one of my feeds to only be posts from artist accounts. So I have a database table of handles, which has a foreign key to a feeds table, and I have user1.bsky,social ... user27.bsky,social in the handles table (which also has a DID column that gets filled automatically on saving the new record), and they each are assigned a foreign key ID of 1 in the feed column (which goes to, e.g., "Artists Feed") in the feeds table.

In `data_filter.py` so far I haven't found a way to extract the DID of the creator from each post being examined, so I can check to see if the post creator DID is A) in the handles table and B) is assigned feed number 1, 2, 3 or so forth.

**Reply by MarshalX** [reply]
@venteto  I'm a bit upset that you asked in reply about "more than one feed behind single hostname" instead of creating a dedicated question in the discussion... but the answer in post uri. AtUri to be more precise. it contains did of the post owner. to work with this uri you should use this tool: https://atproto.blue/en/latest/atproto_core/uri.html

code snippet:
```python
from atproto import AtUri
# `created_post` exists in `data_filter.py` example repo
uri = created_post['uri']
author_did = AtUri.from_str(uri).hostname

# after that you can resolve this did using did resolver but pls do it not in the filter because it performs requests and could be slow
from atproto import IdResolver
resolver = IdResolver()  # you should pass cache here too
print(resolver.did.resolve(author_did))
```

pls next time create separated question

**Reply by venteto** [reply]
My apologies, I was just trying not to pollute Discussions with a lot of my dumb questions, as I imagine everyone else here is smarter than me and rolling their eyes at my dumb questions

**Reply by MarshalX** [reply]
> My apologies, I was just trying not to pollute Discussions with a lot of my dumb questions, as I imagine everyone else here is smarter than me and rolling their eyes at my dumb questions

Maybe they are shy to ask and I really think that your questions are a great contribution to others  ✨

**Comment by venteto** [comment]
Thanks for the hints, I will experiment some more with this when I get a chance
